lib/engine/tile.rb

```diff
@@ -281,7 +281,7 @@ def reserved_by?(corporation)
 
     def add_reservation!(entity, city, slot = 0)
       # Single city, assume the first
-      city = 0 if @cities.one?
+      city = 0 if @cities.one? && available_slot?
```

Comment:
I have an itch on this change. The point is that if the city is full, you should place the reservation on the tile.

Without this change, its possible to have a reservation on the same slot as a token and actually replace that token.

I'm wondering if instead it would be better to make a slightly more verbose function.

```
def add_reservation!(entity, city, slot = nil)
  city = 0 if @cities.one?
  slot = @cities[city].get_slot(entity) if slot.nil?

  if city && slot
    @cities[city].add_reservation!(entity, slot)
  else
    @reservations << entity
  end
end

Reply:
doesn't that change behavior since you're no longer calling available_slot?

prefer unless slot over if slot.nil?

Reply:
Aside: So far this change works fine and none of the tests are breaking. It just has a bad smell to me.

Reply:
why not do slot = available_slot unless slot

Reply:
> doesn't that change behavior since you're no longer calling available_slot?

The later if statement `if city && slot` handles whether there is a slot just as the PR does, but it _seems_ improved in a multiple city case. It may be irrelevant as this may never come up.